mne/time_frequency/tfr.py

```diff
@@ -1512,8 +1512,14 @@ def plot_joint(self, timefreqs=None, picks=None, baseline=None,
             fmax = freq + freq_half_range
 
             data = tfr.data
-
-            pos = find_layout(tfr.info).pos if layout is None else layout.pos
+			
+            if layout is None:
+                layout = find_layout(tfr.info)
+            
+            # only use position information for channels from layout 
+            # whose names appear as a substring in info['ch_names']
+            idx = [ix for ix, ch_name in enumerate(layout.names) if any([ch_name in ch_name_tfr for ch_name_tfr in tfr.ch_names])]
```

Comment:
break the line or you'll get a pep8 error. e.g., before the if

Reply:
I'd use a generator expression for the `any` clause instead of a list comp maybe.

Reply:
Will do!